src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpSourceGenerators.cs

```diff
@@ -50,5 +53,42 @@ public static void Main()
             Assert.Equal($"{HelloWorldGenerator.GeneratedClassName}.cs {ServicesVSResources.generated_suffix}", VisualStudio.Shell.GetActiveWindowCaption());
             Assert.Equal(HelloWorldGenerator.GeneratedClassName, VisualStudio.Editor.GetSelectedText());
         }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SourceGenerators)]
+        public void FindReferencesForFileWithDefinitionInSourceGeneratedFile()
+        {
+            VisualStudio.Editor.SetText(@"using System;
+internal static class Program
+{
+    public static void Main()
+    {
+        Console.WriteLine(" + HelloWorldGenerator.GeneratedClassName + @".GetMessage());
+    }
+}");
+
+            VisualStudio.Editor.PlaceCaret(HelloWorldGenerator.GeneratedClassName);
+            VisualStudio.Editor.SendKeys(Shift(VirtualKey.F12));
+
+            string programReferencesCaption = $"'{HelloWorldGenerator.GeneratedClassName}' references";
+            var results = VisualStudio.FindReferencesWindow.GetContents(programReferencesCaption);
```

Comment:
can we have tests of this that don't have to go through the entire integration test framework?  i don't mind having this.  but it would be good to have a low level API/feature test of this as well.

Reply:
Agreed, unit tests would be good too. The integration test is valuable too since it also tests OOP, which was what actually needed fixing here.